### DISCUSSION TITLE
type-safe server-side redirect in `typedRoutes`

### DISCUSSION THREAD

**Discussion by JoseVSeb** [vercel/next.js#49393] [App Router]
Experimental feature `typedRoutes` currently gives type-safety only to client-side navigation like `useRouter` and `Link`. Extend the same to server-side redirects as well. like `redirect` from `next/navigation` and `NextResponse.redirect(...)`.

**Comment by PedroAVJ** [comment]
Bump. Should this be made an issue so it is tracked?

**Comment by pepew-le-boss** [comment]
I think this should be an issue as this is still not typed
